src/components/Recipe.js

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from "react";
+import { Link, useParams, useHistory } from "react-router-dom";
+import {
+  Title,
+  Button,
+  CircleList,
+  CircleListItem,
+  Container,
+} from "./Styles.js";
+import api from "../data/DataAPI.js";
+
+function IngredientList(props) {
+  const { ingredients } = props;
+  return (
+    <CircleList>
+      {ingredients &&
+        ingredients.map((ingredient) => (
+          <CircleListItem key={ingredient.name}>
+            {ingredient.name}
+          </CircleListItem>
+        ))}
+    </CircleList>
+  );
+}
+
+function RecipeDetail({ recipe }) {
+  const history = useHistory();
+
+  const deleteRecipe = () => {
+    let url = `${process.env.REACT_APP_API_BASEURL}/api/recipe/recipes/${recipe.id}/`;
+
+    fetch(url, { method: "DELETE" }).then(() => history.push("/recipes/"));
+  };
+
+  return (
+    <div>
+      {recipe ? (
+        <div>
+          <Title data-testid="recipe_title">{recipe.name}</Title>
+          <p data-testid="recipe_description">{recipe.description}</p>
+          <h3>Ingredients</h3>
+          <Container>
+            <IngredientList ingredients={recipe.ingredients} />
+          </Container>
+          <Container>
+            <Link to="/recipes">
+              <Button primary>Recipes</Button>
+            </Link>
+            <Link
+              to={{
+                pathname: `/recipes/${recipe.id}/edit`,
+              }}
+            >
+              <Button>Edit</Button>
+            </Link>
+            <Button onClick={deleteRecipe}>Delete</Button>
+          </Container>
+        </div>
+      ) : (
+        <Title>Recipe not found</Title>
+      )}
+    </div>
+  );
+}
+
+function Recipe(props) {
+  const { recipeId } = useParams();
+
+  const [{ data, error, isLoading }, dispatch] = useState({
+    data: null,
+    error: null,
+    isLoading: false,
+  });
+
+  useEffect(() => {
+    dispatch((state) => ({ ...state, isLoading: true }));
+    api
+      .fetchRecipe(recipeId)
+      .then((data) => dispatch({ data, error: null, isLoading: false }))
+      .catch((error) => dispatch({ data: null, error, isLoading: false }));
+  }, [recipeId]);
+
+  if (error) return <Title>An error ocurred</Title>;
+
+  return (
+    <div>
+      {isLoading ? <p>Loading recipe...</p> : <RecipeDetail recipe={data} />}
```

Comment:
not sure if we need to create a new component here, it is also creating some confusion with the naming, why not:

```
if (isLoading){
   return (<p>Loading recipe...</p>)
}

return (<div>
￼      {recipe ? (
￼        <div>
￼          <Title data-testid="recipe_title">{recipe.name}</Title>

```